src/lib/merkle_ledger_tests/test_mask.ml

```diff
@@ -410,10 +434,17 @@ let%test_module "Test mask connected to underlying Merkle tree" =
       (* test runner *)
       let with_instances f =
         let db = Base_db.create () in
-        let pack = Any_base.T ((module Base_db), db) in
-        let maskable = pack in
+        let maskable = Any_base.T ((module Base_db), db) in
         let mask = Mask.create () in
         f maskable mask
+
+      let with_chain f =
+        with_instances (fun maskable mask ->
+            let attached1 = Maskable.register_mask maskable mask in
+            let pack2 = Any_base.T ((module Mask.Attached), attached1) in
+            let mask2 = Mask.create () in
+            let attached2 = Maskable.register_mask pack2 mask2 in
```

Comment:
Neat: this works because the table of registered children are in `Maskable`, rather than in the maskable instance (which I almost did). Here, it's enough that `pack2` conforms to the Base interface, it doesn't need anything in the interface for `Maskable`.